src/OrleansDockerUtils/DockerSiloResolver.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Docker.DotNet;
+using Docker.DotNet.Models;
+using Microsoft.Orleans.Docker.Models;
+using Orleans.Runtime;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Docker
+{
+    internal class DockerSiloResolver : IDockerSiloResolver
+    {
+        private readonly ConcurrentDictionary<IDockerStatusListener, IDockerStatusListener> subscribers =
+            new ConcurrentDictionary<IDockerStatusListener, IDockerStatusListener>();
+        private readonly string deploymentId;
+        private readonly Logger log;
+        private readonly object updateLock = new object();
+        private readonly DockerClient dockerClient;
+        private List<DockerSiloInfo> silos;
+        private const string LABEL_FILTER = "label";
+
+        public DockerSiloResolver(
+            string deploymentId,
+            DockerClient dockerClient,
+            Func<string, Logger> loggerFactory = null)
+        {
+            this.deploymentId = deploymentId;
+            this.dockerClient = dockerClient;
+            log = loggerFactory?.Invoke(nameof(DockerSiloResolver));
+        }
+
+
+        public async Task Refresh()
+        {
+            var containerList = await dockerClient.Containers.ListContainersAsync(
```

Comment:
can you please replace  var containerList with the actual type, so its more readable.